opensafely/pull.py

```diff
@@ -65,14 +65,12 @@ def main(image="all", force=False, project=None):
     try:
         updated = False
         for image in images:
-            # currently databuilder is not published, so we ignore it when pulling
-            if image == "databuilder":
-                continue
             tag = f"{REGISTRY}/{image}"
             if force or tag in local_images:
                 updated = True
                 print(f"Updating OpenSAFELY {image} image")
-                subprocess.run(["docker", "pull", tag + ":latest"], check=True)
+                version = get_default_version_for_image(image)
+                subprocess.run(["docker", "pull", tag + f":{version}"], check=True)
```

Comment:
When we do `opensafely pull --project project.yaml` we pull the names and versions of images from the project.yaml.

If someone had `databuilder:tag` in their project, I think this would replace tag with `v0`.

We should probably only do this default tag dance in none project.yaml invocations?

Reply:
I now realise the previous code had the same issue, and would have forced `latest`

Reply:
Yeah, this is very much the short-term fix. It effectively makes `v0` behave like `latest` for databuilder. At some point before too long we'll bump to `v1` and force the small handful of initial projects off `v0` as that won't be a supported version anymore.

But before we release `v2` of databuilder we'll need to solve this problem properly, however exactly we decide things should work.

As long as the behaviour here is no worse than it currently is for `latest` I think I'm happy.